shared/src/vm/host_env.rs

```diff
@@ -807,6 +812,10 @@ where
     tracing::debug!("tx_update {}, {:?}", key, value);
 
     let key = Key::parse(key).map_err(TxRuntimeError::StorageDataError)?;
+    // VPs must be updated via `tx_update_validity_predicate`
+    if key.is_validity_predicate().is_some() {
+        return Err(TxRuntimeError::CannotDirectlyUpdateVp);
+    }
```

Comment:
actually, maybe this would be a bit nicer

```
    // VP update must be validated
    if key.is_validity_predicate().is_some() {
        tx_add_gas(env, code.len() as u64 * WASM_VALIDATION_GAS_PER_BYTE)?;
        validate_untrusted_wasm(&code).map_err(TxRuntimeError::UpdateVpInvalid)?;
    }
```

with having common validation function with `update_validity_predicate`